tests/basic.py

```diff
@@ -21,29 +22,36 @@
     except OSError:
         pass
 
+https_mode = not argv.http_1_0_only
+
 _, certfile = tempfile.mkstemp()
 
 result = subprocess.run(
     f'openssl req -new -x509 -keyout {certfile} -out {certfile} -days 1 -nodes -subj /C=XX'.split(), capture_output=True)
 result.check_returncode()
 
 HTTPS_SERVER_HOSTNAME = '127.0.0.1'
-HTTP_SERVER_PORT = 60443
-ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
-ssl_context.load_cert_chain(certfile=certfile)
+HTTP_SERVER_PORT = 60443 if https_mode else 60080
+
 httpd = http.server.HTTPServer(
     (HTTPS_SERVER_HOSTNAME, HTTP_SERVER_PORT), http.server.SimpleHTTPRequestHandler)
 httpd.timeout = 1
 httpd.allow_reuse_address = True
-httpd.socket = ssl_context.wrap_socket(httpd.socket, server_side=True)
+if https_mode:
+    ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+    ssl_context.load_cert_chain(certfile=certfile)
+    httpd.socket = ssl_context.wrap_socket(httpd.socket, server_side=True)
 
 httpd_thread = threading.Thread(
     target=lambda httpd: httpd.serve_forever(), args=(httpd,), daemon=True)
 httpd_thread.start()
 
 
 def test_https_server(hostname, port, proxy=None):
-    url = f'https://{hostname}:{port}/404'
+    if https_mode:
```

Comment:
url = f'{server_protocol}...'

Reply:
what about 'scheme'?

Reply:
i thought scheme is more pythonic. anyway either works for me